File: Creational/Lab6/Python/realtor.py
from person import *

class Realtor(Person):
    def __init__(self, name_and_surname, phone, email, years_of_work):
        super().__init__( name_and_surname, phone, email)
        self.years_of_work = years_of_work

    def __str__(self):
        return (f"Realtor:\nId: {self.id}, Name and surname: {self.name_and_surname},\n"
                f"Phone number: {self.phone}, Email: {self.email}, Years of work: {self.years_of_work}")

File: Behavioral/Lab19/Typescript/src/Entities/realtor.ts
import { Customer } from './customer';
import { Person } from './person';
import { RealEstate } from './real_estate';

export class Realtor extends Person {
    public yearsOfWork: number;
    public RealEstates: Array<RealEstate> = [];
    public Customers: Array<Customer> = [];

    constructor(nameAndSurname: string, phone: string, email: string, yearsOfWork: number){
        super(nameAndSurname, phone, email);
        this.yearsOfWork = yearsOfWork;
    }
    public static newRealtorFromObject(realtor: Realtor): Realtor {
        let r = new Realtor(realtor.nameAndSurname, realtor.phone, realtor.email, realtor.yearsOfWork);
        return r;
    }

    public toString(): string {
        return `Realtor:\nId:${this.id}, Name and surname: ${this.nameAndSurname},\nPhone number: ${this.phone}, Email: ${this.email}, Years of work ${this.yearsOfWork}`;
    }

    public NotifyCustomers(): void {
        console.log(`Realtor ${this.nameAndSurname} notified about cheap estate`);
        const cheapEstate = this.RealEstates.sort((a, b) => a.priceInUSD - b.priceInUSD)[0];
        this.Customers.forEach((customer) => {
            customer.notify(`Cheap estate on address ${cheapEstate.address}`);
        });
    }

}